core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -544,28 +544,23 @@ class KafkaApis(val requestChannel: RequestChannel,
 
       // fetch response callback invoked after any throttling
       def fetchResponseCallback(bandwidthThrottleTimeMs: Int) {
-        def createResponse(requestThrottleTimeMs: Int): RequestChannel.Response = {
+        def createResponse(requestThrottleTimeMs: Int): FetchResponse = {
           val convertedData = new util.LinkedHashMap[TopicPartition, FetchResponse.PartitionData]
           fetchedPartitionData.asScala.foreach { case (tp, partitionData) =>
             convertedData.put(tp, convertedPartitionData(tp, partitionData))
           }
-          val response = new FetchResponse(convertedData, 0)
-          val responseSend = response.toSend(bandwidthThrottleTimeMs + requestThrottleTimeMs, request.connectionId, request.header)
-
-          trace(s"Sending fetch response to client $clientId of ${responseSend.size} bytes.")
```

Comment:
Forgot to respond to this. I made this message generic to all APIs and relocated it in `RequestChannel.sendResponse`.